src/ctest.ts

```diff
@@ -619,6 +647,11 @@ export class CTestDriver implements vscode.Disposable {
 
     private async debugCTestHelper(tests: vscode.TestItem[], run: vscode.TestRun, cancellation: vscode.CancellationToken): Promise<number> {
         let returnCode: number = 0;
+
+        if (!await this.checkTestPreset(tests)) {
```

Comment:
Is there a way to know at this point whether presets are being used?  We shouldn't check this if you're not using presets.

Reply:
`useCMakePresets` in runCTest* was added by Elah. Now that you say this, I found it has some problems. There's only one test explorer instance, and all folders share it. So userCMakePresets shouldn't be passed in, or there will be issues when trying to run tests in a preset folder and kit folder at the same time. I'll fix this.

As for this function itself, it checks if there are tests with magic name `_test_preset_required_`. And it shouldn't be added to the test list if the folder is not using CMake presets. I'll use a better name, like "testsRequireTestPreset", and add a comment.